monkey/infection_monkey/monkey.py

```diff
@@ -486,6 +491,11 @@ def _stop_relay(self):
             if self._control_channel.should_agent_stop():
                 self._relay.join(timeout=60)
 
+    def _publish_agent_shutdown_event(self):
+        agent_shutdown_event = AgentShutdownEvent(source=self._agent_id, timestamp=time.time())
+        self._agent_event_queue.publish(agent_shutdown_event)
+        self._island_api_client.send_events([agent_shutdown_event])
```

Comment:
So the event will be sent twice? Once forwarded and once sent explicitly?

Reply:
I think we need 2 things.

1: This should be published and we should allow the `AgentEventForwarder` to handle sending to the island.
2. We should add a `flush()` method to `AgentEventForwarder` and call it before we call `self._close_tunnel()`

We can do #2 in another PR.

Reply:
Dropped the commit that was sending the event explicitly